### PR TITLE
feat: Add minting for authorized accounts

### PR DESCRIPTION
**Pull request by DoubleOTheven**
Resolves #

- [ ] There is an associated issue (**required**)
- [x] The change is described in detail
- [x] There are new or updated tests validating the change (if applicable)

## Description

Add `mintFor` and `updateProfileFor` functions for contracts that use OpenZepplin's Ownership or AccessControl -> DEFAULT_ADMIN_ROLE contracts. This allows for backwards compatibility and ease of profile creation for contracts that may want a Profile.

### CODE REVIEW

File: contracts/moda-contracts/src/Profile.sol [hunk 1]
```diff
@@ -5,21 +5,50 @@ import {IERC165, ERC165} from "@openzeppelin/contracts/utils/introspection/ERC16
...
 
+    bytes32 private constant _DEFAULT_ADMIN_ROLE = 0x00;
+
     error ProfileAlreadyMinted();
     error ProfileDoesNotExist();
     error ProfilesAreSoulBound();
+    error CallerNotAuthorized();
+
+    /// @dev Check if the caller is an owner or admin of a contract.
+    /// @param kontract The address of a contract that may or may not use OpenZeppelin's Ownership
```
Comment (MichaelGrinstead) [inline contracts/moda-contracts/src/Profile.sol:29]: spelling "kontract"

File: contracts/moda-contracts/src/Profile.sol [hunk 2]
```diff
@@ -40,6 +69,20 @@ contract Profile is IProfile, IERC721, IERC721Metadata, IERC4906, ERC165 {
         emit Transfer(address(this), msg.sender, totalSupply);
     }
 
+    /**
+     * @dev See {IProfile-mintFor}.
+     */
+    function mintFor(address kontract, string memory uri) external requireAuthority(kontract) {
```
Comment (MichaelGrinstead) [inline contracts/moda-contracts/src/Profile.sol]: kontract is intentional spelling?

**Review by MichaelGrinstead — APPROVED** [review]
Cool, looks good.

File: contracts/moda-contracts/src/Profile.sol (hunk 2 above)
Comment (bitcoinbrisbane) [inline contracts/moda-contracts/src/Profile.sol]: uri should be calldata not memory too.

File: contracts/moda-contracts/src/Profile.sol [hunk 3]
```diff
@@ -52,6 +95,18 @@ contract Profile is IProfile, IERC721, IERC721Metadata, IERC4906, ERC165 {
         emit MetadataUpdate(tokenId);
     }
 
+    /**
+     * @dev See {IProfile-updateProfileFor}.
+     */
+    function updateProfileFor(address kontract, string memory uri) external requireAuthority(kontract) {
```
Comment (bitcoinbrisbane) [inline contracts/moda-contracts/src/Profile.sol]: And here https://medium.com/coinmonks/solidity-storage-vs-memory-vs-calldata-8c7e8c38bce

File: contracts/moda-contracts/src/Profile.sol (hunk 2 above)
Comment (DoubleOTheven) [reply contracts/moda-contracts/src/Profile.sol]: ah, yes. thanks

Comment (DoubleOTheven) [reply contracts/moda-contracts/src/Profile.sol]: yes, contract is a reserved word

**Pull request merged**